components/src/preact/mutationFilter/mutation-filter.tsx

```diff
@@ -166,151 +245,73 @@ function getPlaceholder(referenceGenome: ReferenceGenome) {
-        </>
-    );
-};
-
-const SelectedAminoAcidInsertion: FunctionComponent<{
-    insertion: InsertionClass;
-    onDelete: (insertion: InsertionClass) => void;
-}> = ({ insertion, onDelete }) => {
-    const backgroundColor = singleGraphColorRGBByName('teal', 0.3);
-    const textColor = singleGraphColorRGBByName('teal', 1);
-    return (
-        <SelectedFilter
-            mutation={insertion}
-            onDelete={onDelete}
-            backgroundColor={backgroundColor}
-            textColor={textColor}
-        />
-    );
-};
-
-const SelectedAminoAcidMutation: FunctionComponent<{
-    mutation: SubstitutionClass | DeletionClass;
-    onDelete: (mutation: SubstitutionClass | DeletionClass) => void;
-}> = ({ mutation, onDelete }) => {
-    const backgroundColor = singleGraphColorRGBByName('rose', 0.3);
-    const textColor = singleGraphColorRGBByName('rose', 1);
-    return (
-        <SelectedFilter
-            mutation={mutation}
-            onDelete={onDelete}
-            backgroundColor={backgroundColor}
-            textColor={textColor}
-        />
-    );
-};
-
-const SelectedNucleotideMutation: FunctionComponent<{
-    mutation: SubstitutionClass | DeletionClass;
-    onDelete: (insertion: SubstitutionClass | DeletionClass) => void;
-}> = ({ mutation, onDelete }) => {
-    const backgroundColor = singleGraphColorRGBByName('indigo', 0.3);
-    const textColor = singleGraphColorRGBByName('indigo', 1);
-    return (
-        <SelectedFilter
-            mutation={mutation}
-            onDelete={onDelete}
-            backgroundColor={backgroundColor}
-            textColor={textColor}
-        />
-    );
-};
-
-const SelectedNucleotideInsertion: FunctionComponent<{
-    insertion: InsertionClass;
-    onDelete: (insertion: InsertionClass) => void;
-}> = ({ insertion, onDelete }) => {
-    const backgroundColor = singleGraphColorRGBByName('green', 0.3);
-    const textColor = singleGraphColorRGBByName('green', 1);
-
-    return (
-        <SelectedFilter
-            mutation={insertion}
-            onDelete={onDelete}
-            backgroundColor={backgroundColor}
-            textColor={textColor}
-        />
+        </div>
     );
 };
 
-type SelectedFilterProps<MutationType extends MutationClass> = {
-    mutation: MutationType;
-    onDelete: (mutation: MutationType) => void;
-    backgroundColor: string;
-    textColor: string;
+type SelectedFilterProps = {
+    handleRemoveValue: (mutation: ParsedMutationFilter) => void;
+    mutationFilter: ParsedMutationFilter;
 };
 
-const SelectedFilter = <MutationType extends MutationClass>({
-    mutation,
-    onDelete,
-    backgroundColor,
-    textColor,
-}: SelectedFilterProps<MutationType>) => {
+const SelectedFilter = ({ handleRemoveValue, mutationFilter }: SelectedFilterProps) => {
     return (
-        <span
-            class='inline-block mx-1 px-2 py-1 font-medium text-xs rounded-full'
-            style={{ backgroundColor, color: textColor }}
+        <button
```

Comment:
In conjunction with the other change (dedicated delete button), we don't need a button here.
```suggestion
        <span
```